### PR TITLE
plumber: init at 2.5.2

### LINKED ISSUE #274259: Package request: plumber
**Project description**

> a CLI devtool for inspecting, piping, messaging and redirecting data in message systems like Kafka, RabbitMQ , GCP PubSub and many more.

https://github.com/streamdal/plumber

**Metadata**

* homepage URL: https://github.com/streamdal/plumber
* source URL: https://github.com/streamdal/plumber
* license: MIT
* platforms: unix

- has a brew recipe already, could serve as inspiration
- it's a Go package

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by svrana**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by svrana** [comment]
requested in #274259

File: pkgs/by-name/pl/plumber/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchFromGitHub {
+    owner = "streamdal";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-ftXLipJQjRdOSNO56rIRfAKKU0kHtAK85hgcT3nYOKA=";
+  };
+
+  vendorHash = null;
+  doCheck = false; # connection tests create a config file in user home directory which fails in the sandbox
```
Comment (svrana) [inline pkgs/by-name/pl/plumber/package.nix:15]: DIdn't see a way to skip just a directory of tests.. not sure how frowned upon this is.

**Comment by svrana** [comment]
Result of `nixpkgs-review pr 274767` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>plumber</li>
  </ul>
</details>

File: pkgs/by-name/pl/plumber/package.nix (hunk 1 above)
Comment (a-n-n-a-l-e-e) [reply pkgs/by-name/pl/plumber/package.nix:15]: can try something like:
```nix
  preCheck = ''
    export HOME="$(mktemp -d)"
  '';
```

Comment (svrana) [reply pkgs/by-name/pl/plumber/package.nix]: It worked. Thanks @a-n-n-a-l-e-e!

File: pkgs/by-name/pl/plumber/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  '';
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/streamdal/plumber/options.VERSION=${version}"
+  ];
+
+  meta = with lib; {
+    description = "plumber is a CLI devtool for inspecting, piping, messaging and redirecting data in message systems like Kafka, RabbitMQ, GCP PubSub and more";
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/pl/plumber/package.nix:28]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes
description can not start with package name and perhaps could be shorter.

File: pkgs/by-name/pl/plumber/package.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  # connection tests create a config file in user home directory
+  preCheck = ''
+    export HOME="$(mktemp -d)"
+  '';
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/streamdal/plumber/options.VERSION=${version}"
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/pl/plumber/package.nix:24]: should this set the version somehow? it doesn't seem to appear when using the command line.
```console
$ ./result/bin/plumber --version
UNSET
```

**Review by ghost — COMMENTED** [review]
looks good. meta.description needs some minor updates to fit with the guidelines and the `--version` flag doesn't seem to show the version.


builds fine on x64 linux and darwin.

File: pkgs/by-name/pl/plumber/package.nix (hunk 3 above)
Comment (svrana) [reply pkgs/by-name/pl/plumber/package.nix:24]: I took this from the upstream Makefile. It also does not work there and I'm not sure why, tried some alternatives, i.e, just options.VERSION but also didn't work, though seems to me both should work just fine. :(

Comment (a-n-n-a-l-e-e) [reply pkgs/by-name/pl/plumber/package.nix:24]: [go.mod](https://github.com/streamdal/plumber/blob/90ee8c815f6dec674dd51602332012d83827f4f5/go.mod#L1C1-L1C36) sets `module github.com/batchcorp/plumber` so need to use `-X github.com/batchcorp/plumber/options.VERSION=${version}`

Comment (a-n-n-a-l-e-e) [reply pkgs/by-name/pl/plumber/package.nix:24]:
```suggestion
    "-X github.com/streamdal/plumber/options.VERSION=${version}"
    # remove once module in go.mod is renamed to github.com/batchcorp/streamdal
    "-X github.com/batchcorp/plumber/options.VERSION=${version}"

```

**Review by ghost — APPROVED** [review]
Result of `nixpkgs-review pr 274767` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>plumber</li>
  </ul>
</details>

```console
./results/plumber/bin/plumber --version
2.5.2
```

**Comment by kubukoz** [comment]
Thanks for fulfilling the package request!

**Pull request merged**